gen/enum.go

```diff
@@ -125,6 +125,23 @@ func enum(g Generator, spec *compile.EnumSpec) error {
 			return []byte(<$strconv>.FormatInt(int64(<$v>), 10)), nil
 		}
 
+		<$zapcore := import "go.uber.org/zap/zapcore">
+
+		// MarshalLogObject implements zapcore.ObjectMarshaler, enabling
+		// fast logging of <$enumName>.
+		func (<$v> <$enumName>) MarshalLogObject(enc <$zapcore>.ObjectEncoder) error {
+			enc.AddInt32("value", int32(<$v>))
+			<if len .Spec.Items ->
+				switch int32(<$v>) {
+				<range .UniqueItems ->
+					case <.Value>:
+						enc.AddString("name", "<.Name>")
+				<end ->
```

Comment:
I assume that there's no `default` case in the switch because there's no safe default name, correct? If so, can we leave a comment in the generated code explaining that?

Reply:
+1

Reply:
Quick update: Comments in the code aren't propagated by the templating logic
we have here. Godocs are propagated, though, so @mh-park added a comment about
how enums are serialized as a godoc.